.github/workflows/phpunit-tests.yml

```diff
@@ -47,64 +38,11 @@ jobs:
         with:
           access_token: ${{ github.token }}
 
-      - name: Checkout repository
-        uses: actions/checkout@v2
-
-      - name: Log debug information
-        run: |
-          echo "$GITHUB_REF"
-          echo "$GITHUB_EVENT_NAME"
-          npm --version
-          node --version
-          curl --version
-          git --version
-          svn --version
-          php --version
-          php -i
-          locale -a
-
-      - name: Install NodeJS
-        uses: actions/setup-node@v2
-        with:
-          node-version: 14
-
-      - name: Cache NodeJS modules
-        uses: actions/cache@v2
-        env:
-          cache-name: cache-node-modules
-        with:
-          # npm cache files are stored in `~/.npm` on Linux/macOS
-          path: ~/.npm
-          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.os }}-npm-
-
-      - name: Install Dependencies
-        run: npx install-changed --install-command="npm ci"
```

Comment:
Is it worth leaving this line in the setup job? If no `npm` cache is present when a workflow starts, every job spun off in the matrix below will need to perform a fresh install of all NPM dependencies. But leaving it here could prime the cache for all of the jobs below.

Reply:
Yes good point. I'll add it back in, and add the `needs:` back to the test jobs.

Reply:
https://core.trac.wordpress.org/ticket/52660 was opened to explore this.